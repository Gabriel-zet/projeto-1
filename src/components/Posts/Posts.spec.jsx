import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'title1',
      body: 'body1',
      cover: 'img/img.png',
    },

    {
      id: 2,
      title: 'title 2',
      body: 'body 2',
      cover: 'img/img2.png',
    },
    {
      id: 3,
      title: 'title 3',
      body: 'body 3',
      cover: 'img/img3.png',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: props.cover })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  it('should not render posts ', () => {
    render(<Posts />);
    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
  });

  it('Should match a snapshot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
