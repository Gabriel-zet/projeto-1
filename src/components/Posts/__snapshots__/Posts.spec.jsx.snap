// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> Should match a snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <h1>
      1
    </h1>
    <img
      alt="title1"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title1
      </h2>
      <p>
        body1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <h1>
      2
    </h1>
    <img
      alt="title 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 2
      </h2>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <h1>
      3
    </h1>
    <img
      alt="title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 3
      </h2>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
